### PR TITLE
fix(smoketest): fail early on unhealthy RHDH startup

### PR DESCRIPTION
**Pull request by rostalan**
Add a startup log gate after health checks so smoke tests fail on backend boot/config errors instead of only plugin load symptoms. This makes overlay smoke failures clearer and diagnostics easier.

### CODE REVIEW

File: .github/workflows/run-workspace-smoke-tests.yaml [hunk 1]
```diff
@@ -149,8 +171,8 @@ jobs:
               failures+=("$plugin")
             fi
           done
-          if echo "$LOGS" | grep -E "(InstallException|Error while adding OCI plugin|Failed to load dynamic plugin|dynamic plugin.*error)" >/dev/null; then
-            echo "Detected dynamic plugin loading errors in logs"
+          if echo "$LOGS" | grep -qiE "$SHARED_HARD_ERROR_PATTERN"; then
```
Comment (gashcrumb) [inline .github/workflows/run-workspace-smoke-tests.yaml]: When is this particular case hit?  Seems like normally the startup check would always catch this first, wouldn't it?

Comment (rostalan) [reply .github/workflows/run-workspace-smoke-tests.yaml]: I guess there could be an edge case, where the log is printed in between, but since the jobs run very fast, it makes little sense to check it twice in the span of <10 seconds. 
-> merged the two steps into the second one.

**Review by gashcrumb — APPROVED** [review]
